src/components/EventDescription.js

```diff
@@ -0,0 +1,263 @@
+        !currFavStatus
+      );
+      setEventFavStatus(!currFavStatus);
+      openSnackBar(favResult);
+    } catch (error) {
+      openSnackBar(error);
+    }
+  };
+
+  const handleClickRegisterOrUnRegisterEvent = async (
+    userID,
+    eventID,
+    isRegister
+  ) => {
+    let isFirstTime = false;
+    registration ? (isFirstTime = false) : (isFirstTime = true); //if registration prop is not undefined, the event has been registered / unregistered before
+    
+    try {
+      const registrationResult = await sendRegistrationData(
+        userID,
+        eventID,
+        isRegister,
+        isFirstTime
+      );
+      setEventRegistrationStatus(isRegister);
+      openSnackBar(registrationResult);
+    } catch (error) {
+      openSnackBar(error);
+    }
+  };
+
+  const openSnackBar = msg => {
+    setSnackMsg(msg);
+    setSnackOpen(true);
+    setTimeout(() => {
+      setSnackOpen(false);
+    }, 1000);
+  };
+
+  return (
+    <React.Fragment>
+      <Paper className={classes.paper}>
+        <div style={{ position: "relative" }}>
+          <Typography variant="h1" className={classes.title}>
+            {event.id}
+          </Typography>
+          <VisibilityIcon
+            className={classes.viewLogo}
+            fill="none"
+          ></VisibilityIcon>
+          {eventFavStatus ? (
+            <StarIcon
+              className={classes.favLogo}
+              onClick={() => {
+                handleClickFavouriteEvent(user.id, event.id);
+              }}
+            ></StarIcon>
+          ) : (
+            <StarBorderOutlinedIcon
+              className={classes.favLogo}
+              onClick={() => {
+                handleClickFavouriteEvent(user.id, event.id);
+              }}
+            ></StarBorderOutlinedIcon>
+          )}
+        </div>
+        <Markdown className={classes.description}>{event.description}</Markdown>
+        <div className={classes.buttonGroup}>
+          {eventRegistrationStatus ? (
+            <React.Fragment>
+              <Button
+                style={{ backgroundColor: COLOR.LIGHT_BACKGROUND_COLOR }}
+                className={classes.button}
+                onClick={() => {
+                  handleClickRegisterOrUnRegisterEvent(user.id, event.id, false)
+                }}
+              >
+                Unregiseter
+              </Button>
+              <Button
+                disabled
+                style={{ backgroundColor: COLOR.BIZTECH_GREEN }}
+                className={classes.button}
+              >
+                Registered
+              </Button>
+            </React.Fragment>
+          ) : (
+            <Button
+              style={{ backgroundColor: COLOR.BIZTECH_GREEN }}
+              className={classes.button}
+            >
+              sign me up
+            </Button>
+          )}
+        </div>
+      </Paper>
+      <Snackbar
+        open={snackOpen}
+        TransitionComponent={TransitionUp}
+        message={snackMsg}
```

Comment:
is there a way to set the snackbar message without creating a state variable? there might be a helper function that you could use in the same place as setSnackOpen and setSnackMsg but i'm not sure